### PR TITLE
Adding Oauth2 and fixing errors and warnings

### PR DESCRIPTION
**Pull request by JoshL579**

### CODE REVIEW

File: client/src/pages/Login.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            .then(res => {
+                status = res.status;
+                if (status < 500) return res.json();
+                else throw Error("Server error");
+            })
+            .then(res => {
+                if (status === 200) alert(res.response);
+                const cookies = new Cookies();
+                cookies.set('token', res.token, { path: '/', httpOnly: true })
+                if (status === 401) alert(res.response);
```
Comment (johnsciutto) [inline client/src/pages/Login.js:79]: Hi! I'm assuming you'll change these `alert`(s) in the future to do something else with the response?

File: client/src/pages/Login.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                        <Button className={classes.btnOrange}
+                            onClick={handleGetStarted}>Continue</Button>
+                    </CardContent>
+                )}
+                {confirmed === true && (
+                    <CardContent className={classes.body}>
+                        <Typography variant="h6" className={classes.title}>
+                            Welcome back,
+                                <br />{email}
+                        </Typography>
```
Comment (johnsciutto) [inline client/src/pages/Login.js]: Small thing, but you could probably split these two lines into separate `<Typography>` components, and you wouldn't have to use the `<br/>`.

File: client/src/pages/Login.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                        <CardActions>
+                            <GoogleLogin
+                                clientId="802130452785-qc5dp590pd2qs6opc6udqa8qt6ofdihl.apps.googleusercontent.com"
```
Comment (johnsciutto) [inline client/src/pages/Login.js]: Is this supposed to be out in the open? It looks like something that you would want in the `.env` file.

**Review by johnsciutto — COMMENTED** [review]
There seems to be a lot of repeated code between the **client/src/pages/Login.js** and **client/src/pages/SignUp.js** files. See if you can abstract some of the logic/components/styles to have less repetition between them.

File: client/src/pages/Login.js (hunk 1 above)
Comment (JoshL579) [reply client/src/pages/Login.js:79]: Exactly!

**Review by sina-jamshidi — APPROVED** [review]

**Comment by JoshL579** [comment]
Just fixed the problems you guys mentioned above, please check once again, if no problems, I'll then merge.

**Pull request merged**